### PR TITLE
DiskRP swagger for 2021-04-01

### PR DESCRIPTION
**Pull request by rytruong**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [x] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-04-01/disk.json [hunk 1]
```diff
@@ -2470,6 +2480,28 @@
...
+        "Enabled",
+        "Disabled"
+      ],
+      "x-ms-enum": {
+        "name": "PublicNetworkAccess",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "Enabled",
+            "description": "Export on disk will depend on the networkAccessPolicy"
```
Comment (chetmaddula) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2021-04-01/disk.json:2496]: needs better description.. will send it in the email

**Comment by chiragg4u** [comment]
I don't see any changes in the swagger in the new version, please confirm if it's ready for review?

**Comment by chetmaddula** [comment]
@chiragg4u https://github.com/Azure/azure-rest-api-specs/pull/14798/files/34833b38dfd43886fd7753c052cff376c9823d01..1fe26a2d52241fed2a0317504e3581f525a976b9
This link would have the changes for swagger. Its in disk.json
@rytruong can you verify why the diff is showing something being removed from compute.json

**Review by haagha — COMMENTED** [review]
Do we have an example for the new Clone option?

File: specification/compute/resource-manager/readme.md [hunk 2]
```diff
@@ -238,7 +238,7 @@ input-file:
 - Microsoft.Compute/stable/2021-03-01/compute.json
 - Microsoft.Compute/stable/2021-03-01/runCommands.json
 - Microsoft.Compute/stable/2019-04-01/skus.json
-- Microsoft.Compute/stable/2020-12-01/disk.json
+- Microsoft.Compute/stable/2021-04-01/disk.json
```
Comment (Sandido) [inline specification/compute/resource-manager/readme.md:241]: This is the first time 2021-04 is being introduced to Swagger, it cannot remain in the 2021-03-01 tag. Please make a new Tag for the 2021-04 release, and update the line at the top of this doc "tag: package-2021-03-01" to 2021-04.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-04-01/disk.json [hunk 3]
```diff
@@ -0,0 +1,3605 @@
...
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ShareInfoElement"
+          },
+          "description": "Details of the list of all VMs that have the disk attached. maxShares should be set to a value greater than one for disks to allow attaching them to multiple VMs."
+        },
+        "networkAccessPolicy": {
+          "$ref": "#/definitions/NetworkAccessPolicy"
+        },
+        "publicNetworkAccess": {
```
Comment (Sandido) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2021-04-01/disk.json:2122]: Add new optional properties at the end of the property list, not in the middle.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-04-01/disk.json [hunk 4]
```diff
@@ -0,0 +1,3605 @@
...
+          "type": "boolean",
+          "description": "Indicates the OS on a disk supports hibernation."
+        },
+        "securityProfile": {
+          "$ref": "#/definitions/DiskSecurityProfile",
+          "description": "Contains the security related information for the resource."
+        },
+        "completionPercent": {
+          "type": "number",
+          "description": "Percent complete of a clone operation"
```
Comment (Sandido) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2021-04-01/disk.json:2152]: add a period at the end, '.'

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-04-01/disk.json [hunk 5]
```diff
@@ -0,0 +1,3605 @@
...
+          "description": "Whether a snapshot is incremental. Incremental snapshots on the same disk occupy less space than full snapshots and can be diffed."
+        },
+        "encryption": {
+          "$ref": "#/definitions/Encryption",
+          "description": "Encryption property can be used to encrypt data at rest with customer managed keys or platform managed keys."
+        },
+        "networkAccessPolicy": {
+          "$ref": "#/definitions/NetworkAccessPolicy"
+        },
+        "publicNetworkAccess": {
```
Comment (Sandido) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2021-04-01/disk.json:2240]: add new properties to the end of property lists, not in the middle

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-04-01/disk.json [hunk 6]
```diff
@@ -0,0 +1,3605 @@
...
+        "Enabled",
+        "Disabled"
+      ],
+      "x-ms-enum": {
+        "name": "PublicNetworkAccess",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "Enabled",
+            "description": "Export on disk will depend on the networkAccessPolicy"
```
Comment (Sandido) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2021-04-01/disk.json:2496]: Need a period at the end of descriptions.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-04-01/disk.json [hunk 7]
```diff
@@ -0,0 +1,3605 @@
...
+                "value": "Restore",
+                "description": "Create a new disk by copying from a backup recovery point."
+              },
+              {
+                "value": "Upload",
+                "description": "Create a new disk by obtaining a write token and using it to directly upload the contents of the disk."
+              },
+              {
+                "value": "Clone",
+                "description": "Create a new disk by using a slow/deep copy process, where the resource creation is considered complete only after all data has been copied from the source."
```
Comment (Sandido) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2021-04-01/disk.json:2741]: Create a new disk using a deep copy process

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-04-01/disk.json [hunk 8]
```diff
@@ -0,0 +1,3605 @@
...
+        "publicNetworkAccess": {
+          "$ref": "#/definitions/PublicNetworkAccess"
+        },
+        "diskAccessId": {
+          "type": "string",
+          "description": "ARM id of the DiskAccess resource for using private endpoints on disks."
+        },
+        "completionPercent": {
+          "type": "number",
+          "description": "Percent complete of a clone operation"
```
Comment (Sandido) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2021-04-01/disk.json:3392]: add a period at the end of the description

**Review by Sandido — CHANGES_REQUESTED** [review]

**Review by Sandido — CHANGES_REQUESTED** [review]

**Comment by chiragg4u** [comment]
Thank you for copying the existing version into new folder for first commit. It helps in reviewing the change.
Changes looks good from ARM perspective, signedoff. Please resolve other warnings/errors and follow the process given @ https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/212/Swagger-PR-Review to continue

**Comment by yungezz** [comment]
hi @rytruong could you rebase the PR, and fix CI failure? Will review again once ready.

**Comment by grizzlytheodore** [comment]
@rytruong  please provide update on this PR

**Comment by chetmaddula** [comment]
> @rytruong please provide update on this PR

rytruong is currently oof. Please contact vikramd and diskrpdev

**Pull request closed without merging**